src/lib/mod.rs

```diff
@@ -139,23 +139,41 @@ pub fn grab_pointer_set_cursor(conn: &xcb::Connection, root: u32) -> bool {
     false
 }
 
-pub fn grab_escape_key(conn: &xcb::Connection, root: u32) {
+pub fn find_escape_keycode(conn: &xcb::Connection) -> xcb::Keycode {
+    // https://stackoverflow.com/questions/18689863/obtain-keyboard-layout-and-keysyms-with-xcb
+    let setup = conn.get_setup();
+    let cookie = xcb::get_keyboard_mapping(
+        &conn,
+        setup.min_keycode(),
+        setup.max_keycode() - setup.min_keycode() + 1,
+    );
+    let reply = cookie.get_reply().expect("failed to get keyboard mapping");
+
+    let escape_index = reply
+        .keysyms()
+        .iter()
+        .position(|&keysym| keysym == ESC_KEYSYM)
+        .expect("failed to find escape keysym");
+    return (escape_index / reply.keysyms_per_keycode() as usize) as u8 + setup.min_keycode();
```

Comment:
minor nit: this 'return' is unneeded since it comes at the end of the function.

Reply:
Oh, I forgot about that. Thanks for reminding, I am surprised clippy didn't mention this.